baremetal/cluster_manager.go

```diff
@@ -78,10 +87,32 @@ func (s *ClusterManager) UpdateConfiguration() error {
 
 // IP returns the cluster manager IP address
 func (s *ClusterManager) APIEndpoints() ([]capbm.APIEndpoint, error) {
+	//Get IP address from spec, which gets it from posted cr yaml
+	// Once IP is handled, consider setting the port
+	endPoint := s.BareMetalCluster.Spec.APIEndpoint
+	// Parse
+	u, err := url.Parse(endPoint)
+	if err != nil {
+		log.Fatalf("Unable to parse IP and PORT from the given url %s", err)
+	}
+	//ip, p, err := net.SplitHostPortAPIEndpointPort(u.Host)
+	ip := u.Hostname()
+	p := u.Port()
+
+	if p == "" {
+		p = APIEndpointPort
+		//log.Fatalf("Unable to split IP and PORT of the given url %v", err)
+	}
+	port, err := strconv.Atoi(p)
+
+	if err != nil {
+		log.Fatalf("Invalid Port %s", err)
+	}
+
 	return []capbm.APIEndpoint{
 		{
-			Host: "",
-			Port: 8080,
+			Host: ip,
+			Port: port,
```

Comment:
when using BareMetalClusterStatus struct as it is, host and port could be:

			Host: s.Cluster.Status.APIEndpoints[0].Host,
			Port: s.Cluster.Status.APIEndpoints[0].Port,